File: source/includes/fts-tutorial/lookup-with-search/query.java
import static com.mongodb.client.model.Aggregates.limit;
import static com.mongodb.client.model.Aggregates.project;
import static com.mongodb.client.model.Projections.*;
import java.util.Arrays;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class LookupWithSearchQuery {

  public static void main(String[] args) {
    // connect to your Atlas cluster
    String uri = "<connection-string>";
        
    try (MongoClient mongoClient = MongoClients.create(uri)) {
      // set namespace
      MongoDatabase database = mongoClient.getDatabase("sample_analytics");
      MongoCollection<Document> collection = database.getCollection("customers");
            
      // define pipeline
      Document agg = new Document("$lookup", 
        new Document("from", "accounts")
          .append("localField", "accounts")
          .append("foreignField", "account_id")
          .append("as", "purchases")
          .append("pipeline", Arrays.asList(new Document("$search", 
            new Document("index", "lookup-with-search-tutorial")
              .append("compound", 
              new Document("must", Arrays.asList(new Document("queryString", 
                new Document("defaultPath", "products")
                  .append("query", "products: (CurrencyService AND InvestmentStock)"))))
                  .append("should", Arrays.asList(new Document("range", 
                    new Document("path", "limit")
                      .append("gte", 5000L)
                      .append("lte", 10000L)
                  )))
              )
          ), 
          new Document("$limit", 5L),
          new Document("$project", 
          new Document("_id", 0L)
            .append("address", 0L)
            .append("birthdate", 0L)
            .append("username", 0L)
            .append("tier_and_details", 0L)
        )))
      );
      // run pipeline and print results
      collection.aggregate(Arrays.asList(agg,
        limit(5),
        project(fields(excludeId(), include("name", "email", "active", "accounts", "purchases")))
      ))
      .forEach(doc -> System.out.println(doc.toJson()));
    }
  }
}

File: source/includes/fts-tutorial/lookup-with-search/query.py
import pymongo

# connect to your Atlas cluster
client = pymongo.MongoClient('<connection-string>')

# define pipeline
pipeline = [
  {
    '$lookup': {
      'from': 'accounts', 
      'localField': 'accounts', 
      'foreignField': 'account_id', 
      'as': 'purchases', 
      'pipeline': [
        {
          '$search': {
            'index': 'lookup-with-search-tutorial', 
            'compound': {
              'must': [
                {
                  'queryString': {
                    'defaultPath': 'products', 
                    'query': 'products: (CurrencyService AND InvestmentStock)'
                  }
                }
              ], 
              'should': [
                {
                  'range': {
                    'path': 'limit', 
                    'gte': 5000, 
                    'lte': 10000
                  }
                }
              ]
            }
          }
        }, 
        { '$project': { '_id': 0 }  }
      ]
    }
  }, 
  { '$limit': 5 }, 
  {
    '$project': {
      '_id': 0, 
      'address': 0, 
      'birthdate': 0, 
      'username': 0, 
      'tier_and_details': 0
    }
  }
]

# run pipeline
result = client['sample_analytics']['customers'].aggregate(pipeline)

# print results
for i in result:
    print(i)
